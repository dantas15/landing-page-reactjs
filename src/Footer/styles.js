import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
