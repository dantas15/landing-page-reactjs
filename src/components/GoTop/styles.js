import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    bottom: 2rem;
    right: 2rem;

    z-index: 6;

    border-radius: 0.4rem;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
  `}
`;
