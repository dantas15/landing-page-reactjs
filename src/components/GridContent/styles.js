import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    text-align: center;
    margin: 0 auto;
  `}
`;

export const Html = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;
  `}
`;
