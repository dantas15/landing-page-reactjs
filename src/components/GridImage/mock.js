export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      altText: 'Teste 1',
      srcImage: 'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      srcImage: 'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      srcImage: 'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      srcImage: 'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      srcImage: 'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      srcImage: 'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
};
