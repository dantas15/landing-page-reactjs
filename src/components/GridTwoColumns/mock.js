export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit, amet consectetur adipisicing elit.
  Repellendus, voluptatibus? Fuga aut et illum voluptates neque
  molestiae consequatur, voluptatibus enim consequuntur delectus
  sed unde ratione velit deleniti eligendi, culpa ipsa?`,
  srcImg: 'assets/images/javascript.svg',
};
