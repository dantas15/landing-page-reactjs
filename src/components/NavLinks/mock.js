export default [
  {
    children: 'Link 1',
    link: '#target1',
    newTab: false,
  },
  {
    children: 'Link 2',
    link: '#target2',
    newTab: false,
  },
  {
    children: 'Link 3',
    link: '#target3',
    newTab: false,
  },
  {
    children: 'Link 4',
    link: '#target4',
    newTab: false,
  },
  {
    children: 'Link 5',
    link: '#target5',
    newTab: false,
  },
];
