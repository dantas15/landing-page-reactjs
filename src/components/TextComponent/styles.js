import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
