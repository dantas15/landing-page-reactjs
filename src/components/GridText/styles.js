import styled, { css } from 'styled-components';

import { Title as HeadingContainer } from 'components/Heading/styles';
import { Container as TextComponent } from 'components/TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;

    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    ${HeadingContainer} {
      position: relative;
      left: 5rem;
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.font.sizes.xhuge};
      top: -3rem;
      left: -5rem;
      transform: rotate(5deg);
    } ;
  `}
`;
