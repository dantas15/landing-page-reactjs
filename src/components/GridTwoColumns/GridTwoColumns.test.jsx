import { screen } from '@testing-library/react';

import { renderTheme } from 'styles/render-theme';

import { GridTwoColumns } from '.';

import mock from './mock';

describe('<GridTwoColumns />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
