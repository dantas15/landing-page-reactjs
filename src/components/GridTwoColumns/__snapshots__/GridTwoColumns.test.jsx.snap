// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render two column grid 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: none;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid Two Columns
          </h2>
          <footer
            class="c6"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit.
  Repellendus, voluptatibus? Fuga aut et illum voluptates neque
  molestiae consequatur, voluptatibus enim consequuntur delectus
  sed unde ratione velit deleniti eligendi, culpa ipsa?
          </footer>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid Two Columns"
            class=""
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
